src/inventory/gilded_rose.js

```diff
@@ -22,49 +22,77 @@ const items = [
 updateQuality(items);
 */
 export function updateQuality(items) {
+    //combine these into quality changer--takes in a + or - number and you can get BOTH increase or decrease in quality
+    qualityUpdater = (item, change_Amount) =>{
+       if (item.quality < 50 && item.quality >0){
```

Comment:
It appears there is some hidden logic in `qualityUpdater` where the quality is only updated if the quality is less and fifty and greater than 0. Is this a strategy for a specific item or for all items?

Reply:
For all items, I have placed this in a different function after some extensive refactor.